RadClass/analysis.py

```diff
@@ -0,0 +1,205 @@
+        If not provided (None), cache_size is ignored (equals integration time).
+    labels: list of dataset name labels in this order:
+        [ live_label: live dataset name in HDF5 file,
+          timestamps_label: timestamps dataset name in HDF5 file,
+          spectra_label: spectra dataset name in HDF5 file ]
+    '''
+
+    running = True
+
+    def __init__(self, stride, integration, datapath, filename, analysis = None,
+                    store_data = False, cache_size = None,
+                                        labels = {'live': '2x4x16LiveTimes',
+                                                  'timestamps': '2x4x16Times',
+                                                  'spectra': '2x4x16Spectra'}):
+        self.stride = stride
+        self.integration = integration
+        self.datapath = datapath
+        self.filename = filename
+
+        self.store_data = store_data
+        if cache_size is None:
+            self.cache_size = self.integration
+        else:
+            self.cache_size = cache_size
+
+        self.labels = labels
+
+        # analysis object that will manipulate data
+        self.analysis = analysis
+
+    def queue_file(self):
+        '''
+        Initialize a file for analysis using the load_data.py scripts.
+        Collects all information and assumes it exists in the file.
+
+        Attributes:
+        processor: DataSet object responsible for indexing data from file.
+        working_time: The current working epoch timestamp at which data is being
+            analyzed. This is used to keep track of progress within a file.
+        '''
+
+        self.processor = ds.DataSet(self.labels)
+        self.processor.init_database(self.filename, self.datapath)
+        # parameters for keeping track of progress through a file
+        
+        self.working_time = self.processor.timestamps[0]
+
+    def collapse_data(self, rows):
+        '''
+        Integrates a subset of data from the total data matrix given some
+        integration time. Utilizes data_slice() from DataSet to extract
+        the requisite rows of data and then numpy.sum(data,axis=0) to combine
+        all rows. Assumes negligible drift and accurate energy calibration.
+        
+        NOTE: Returned data is normalized for live times (1s measurement
+        period less dead time). i.e. The returned data is a count rate at
+        each bin.
+
+        Return: 1D numpy array for integration interval
+        '''
+
+        # extract requisite data rows
+        data_matrix = self.processor.data_slice(self.datapath, rows)
+
+        # normalize by live times to produce count rate data
+        # processor.live can be indexed by appropriate timestamps but
+        # data_matrix only has indices for rows.
+        for idx, row in enumerate(rows):
+            dead_time = self.processor.live[row]
+            data_matrix[idx] = data_matrix[idx] / dead_time
```

Comment:
I get the following error when testing:
```
ValueError: operands could not be broadcast together with shapes (60,1000) (60,)
```
Would it be valid to transpose data_matrix so that the matrix dimensions match? Maybe I could copy the vector self.processor.live[rows] 1000 to create a matrix?

Reply:
Figured it out! Numpy allows you to divide a matrix by a vector (where each row of the matrix is divided by a vector element) when indexed in this way:
```
self.processor.live[rows][:,None]
```